datafusion/sql/src/unparser/utils.rs

```diff
@@ -82,3 +91,28 @@ pub(crate) fn unproject_agg_exprs(expr: &Expr, agg: &Aggregate) -> Result<Expr>
         })
         .map(|e| e.data)
 }
+
+/// Recursively identify all Column expressions and transform them into the appropriate
+/// window expression contained in window.
+///
+/// For example, if expr contains the column expr "COUNT(*) PARTITION BY id" it will be transformed
+/// into an actual window expression as identified in the window node.
+pub(crate) fn unproject_window_exprs(expr: &Expr, window: &Window) -> Result<Expr> {
```

Comment:
I tried to implement `unproject_window_exprs` before by using the same method in `unproject_agg_exprs`, that's finding the `index_of_column` but it's not correct. Could you tell me why there's a difference? 😊

Reply:
The difference I found is that the Aggregate structure maintains all of the relevant Exprs that a Projection node may be referencing in its group by and aggr expr vecs. So, you can look up any column in the schema of the aggregate plan to find the correct expression to sub into the Projection node. 

Window on the other hand only keeps the list of window functions whereas the schema has all of the fields. So its no longer possible to rely on the index of a column within the schema to find the correct expression.

Instead, I am using the auto generated column name for the expression to identify the correct expression. This is definitely a more brittle way to do it and could break if there are unexpected changes to how names are generated or propagated during planning.

Reply:
It turns out another difference is there can be multiple Window nodes representing different WindowSpecs within the same select query. I had to make a few additional tweaks to handle this case.